eth_block.go

```diff
@@ -0,0 +1,362 @@
+package ipldeth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	cid "github.com/ipfs/go-cid"
+	node "github.com/ipfs/go-ipld-format"
+
+	"github.com/ethereum/go-ethereum/common"
+	types "github.com/ethereum/go-ethereum/core/types"
+	rlp "github.com/ethereum/go-ethereum/rlp"
+)
+
+// EthBlock (eth-block, codec 0x90), represents an ethereum block header
+type EthBlock struct {
+	*types.Header
```

Comment:
So, this may just be the last remains of my inner Java programmer but... we're exposing a lot of mutable public fields here on what should be an immutable type. I'm not sure what the best fix is here (or if we should fix this) but getters are an option...